src/Cake.Common/Build/GitLabCI/Data/GitLabCIRunnerInfo.cs

```diff
@@ -47,10 +47,10 @@ public string[] Tags
         {
             get
             {
-                var tags = GetEnvironmentString("CI_RUNNER_TAGS").Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                var tags = GetEnvironmentString("CI_RUNNER_TAGS").Trim('[', ']').Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Question: could the variables contain a `,` within quotes?

Would we need to do something similar to the [QuoteAwareStringSplitter](https://github.com/cake-build/cake/blob/a1893a2bba26f49e089d66beeb78421cedf6126d/src/Cake.Core/Text/QuoteAwareStringSplitter.cs#L14)?

Reply:
I did a test with a self-hosted instance and, although the runner ui let me create a tag with commas, it had to be quoted; but then when editing the tags again they were no longer quoted and got split into multiple tags. So I would argue that such tags are not expected and not really supported. This makes sense given the runner ui states: "You can set up jobs to only use runners with specific tags. Separate tags with commas."

Reply:
Ok, then we'll revisit if it becomes an issue